the_destination_of_shadow/DirectX.cpp

```diff
@@ -78,6 +113,54 @@ HRESULT DirectX::InitD3Device(HWND hWnd, CONST TCHAR* FilePath)
 	return S_OK;
 }
 
+KEY_STATE DirectX::GetInputState(BYTE curr_diks, BYTE prev_diks)
+{
+	if (IsKeyPressed(curr_diks)==TRUE)
+	{
+		return (prev_diks == OFF) ? PRESS : ON;
+	}
+	else
+	{
+		return (prev_diks == ON) ? RELEASE : OFF;
+	}
+}
+
+BOOL DirectX::IsKeyPressed(BYTE curr_diks)
+{
+	return (curr_diks & mask.NUM) ? TRUE : FALSE;
+}
+
+VOID DirectX::CheckInputStateDetails()
+{
+	BYTE curr_diks[KEY_MAX];
+
+	static BYTE prev_diks[KEY_MAX] = {};
+	for (int i = 0; i < KEY_MAX; i++)
```

Comment:
こんなのこそ、一番単純な範囲ベースforの出番では？
```suggestion
	for (BYTE& prev_dik : prev_diks)
```
で良さそう。
(当然、for文の中身は `prev_dik = OFF;` に変えないとダメですが)

Reply:
修正します